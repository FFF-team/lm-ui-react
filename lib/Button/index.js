'use strict';

exports.__esModule = true;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

require('./index.scss');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var propTypes = {
	size: _react2.default.PropTypes.string, //small middle big long
	btnType: _react2.default.PropTypes.string, //primary colorfulHollow  grayHollow
	icon: _react2.default.PropTypes.string,
	isDisabled: _react2.default.PropTypes.bool,
	isRadius: _react2.default.PropTypes.bool,
	onClick: _react2.default.PropTypes.func
};
var defaultProps = {
	size: 'small',
	btnType: 'primary',
	isDisabled: false,
	isRadius: true,
	onClick: function onClick() {}
};

var Button = function (_React$Component) {
	_inherits(Button, _React$Component);

	function Button(props) {
		_classCallCheck(this, Button);

		// debugger;
		return _possibleConstructorReturn(this, _React$Component.call(this, props));
	}

	Button.prototype.clickhandle = function clickhandle(e) {
		this.props.onClick(e.target);
	};

	Button.prototype.render = function render() {
		var _props = this.props,
		    children = _props.children,
		    size = _props.size,
		    btnType = _props.btnType,
		    isRadius = _props.isRadius,
		    isDisabled = _props.isDisabled,
		    onClick = _props.onClick,
		    icon = _props.icon,
		    arg = _objectWithoutProperties(_props, ['children', 'size', 'btnType', 'isRadius', 'isDisabled', 'onClick', 'icon']);

		var classnames = 'lm-ui-btn ';

		switch (btnType) {
			case 'primary':
				classnames += 'lm-ui-btn-primary ';
				break;
			case 'colorfulHollow':
				classnames += 'lm-ui-btn-colorful-hollow ';
				break;
			case 'grayHollow':
				classnames += 'lm-ui-btn-gray-hollow ';
				break;
		}

		if (isRadius) {
			classnames += 'lm-ui-btn-radius ';
		}

		if (isDisabled) {
			classnames += 'lm-ui-btn-disable ';
		}

		switch (size) {
			case 'small':
				classnames += 'lm-ui-btn-small ';
				break;
			case 'big':
				classnames += 'lm-ui-btn-big ';
				break;
			case 'long':
				classnames += 'lm-ui-btn-long ';
				break;
		}

		return _react2.default.createElement(
			'a',
			_extends({ href: 'javascript:;',
				onClick: isDisabled ? undefined : this.clickhandle.bind(this),
				className: classnames
			}, arg),
			_react2.default.createElement('img', { src: icon, className: icon ? 'lm-ui-btn-icon' : 'hide' }),
			children
		);
	};

	return Button;
}(_react2.default.Component);

exports.default = Button;


Button.propTypes = propTypes;
Button.defaultProps = defaultProps;
module.exports = exports['default'];