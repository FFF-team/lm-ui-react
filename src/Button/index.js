import React from 'react';
import './index.scss';

const propTypes = {
	size:React.PropTypes.string,//small middle big long
	type:React.PropTypes.string,//primary colorfulHollow  grayHollow
	icon:React.PropTypes.string,
	isDisabled:React.PropTypes.bool,
	onClick:React.PropTypes.func
};
const defaultProps = {
	size:'small',
	type:'primary',
	isDisabled:false,
	onClick:() => {}
};

export default class Button extends React.Component {

	constructor (props) {
		// debugger;
		super (props);
		this.init();

	}

	clickhandle(e){
		this.props.onClick(e.target);
	}

	init(){
		this.classnames = 'lm-ui-btn ';
		switch(this.props.type){
			case 'primary':
				this.classnames += 'lm-ui-btn-primary ';
				break;
			case 'colorfulHollow':
				this.classnames += 'lm-ui-btn-colorful-hollow ';
				break;
			case 'grayHollow':
				this.classnames += 'lm-ui-btn-gray-hollow ';
				break;
		}

		switch(this.props.size){
			case 'small':
				this.classnames += 'lm-ui-btn-small ';
				break;
			case 'middle':
				this.classnames += 'lm-ui-btn-middle ';
				break;
			case 'big':
				this.classnames += 'lm-ui-btn-big ';
				break;
			case 'long':
				this.classnames += 'lm-ui-btn-long ';
				break;
		}
	}

	render () {

		const { children } = this.props;
		// debugger
		return (

			<button  onClick={this.clickhandle.bind(this)} className={this.classnames} disabled={this.props.isDisabled ? true : false}>
				<img src={this.props.icon} className={this.props.icon ? 'lm-ui-btn-icon' : 'hide'} />
				{ children }
			</button>
		)

	}

}

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;