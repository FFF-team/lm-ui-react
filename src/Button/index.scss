@import '../Style/lmui.scss';


/*通用样式*/
.lm-ui-btn{
	border-radius: 4px;
	font-size: 16px;
	border-width: 1px;
	border-style: solid;
	padding: 12px 20px;
	.lm-ui-btn-icon{
		display: inline-block;
		margin-right: 10px;
		background-size: 100% 100%;
		width: 16px;
		height: 16px;
		vertical-align: middle;
	}
}
.lm-ui-btn[disabled] {
  background-color: #E6E6E6;
  border-color: #E6E6E6;
  color: #FFFFFF;
}

/*尺寸样式*/
.lm-ui-btn-small{
	font-size:14px;
	padding: 6px 10px;
}
.lm-ui-btn-middle{
	font-size:14px;
	padding: 8px 15px;
}
.lm-ui-btn-big{
	font-size:16px;
	padding: 14px 20px;
}
.lm-ui-btn-long{
	display: block;
	width: 100%;
	padding: 12px 0px;
}


/**形态样式**/

	/*普通(实心)*/
.lm-ui-btn-primary{
	color:#FFFFFF;
	background-color: #FF552E;
	border-color: #FF552E;
	&:active{
		background-color:#E04A28;
		border-color: #E04A28;
		border:1px solid #E04A28;
	}
	&:focus{
		border: 1px solid #FF552E;
	}
}
	/*彩色-空心*/
.lm-ui-btn-colorful-hollow{
	color:#FF552E;
	border-color: #FF552E;
	&:active{
		border:1px solid #E04A28;
		color: #E04A28;
	}
	&:focus{
		border: 1px solid #FF552E;
	}
}
	/*灰色-空心*/
.lm-ui-btn-gray-hollow{
	color:#333333;
	border-color: #E6E6E6;
	&:active{
		border:1px solid #E6E6E6;
		background-color: #E6E6E6;
	}
	&:focus{
		border: 1px solid #E6E6E6;
	}
}
  /*吸底*/
.lm-ui-btn-suction{
	position: fixed;
    left: 0;
    right: 0;
    bottom: 0;
    z-index: 99;
    .lm-ui-btn{
    	border-radius: 0px;
    }
    .lm-ui-btn-suction-button{
    	width: 125px;
    }
}
.lm-ui-flex-box{
	display: -webkit-box;
    display: flex;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    .lm-ui-btn-item{
	    -webkit-box-flex: 1;
	    -webkit-flex: 1;
	    -moz-flex: 1;
	    flex: 1;
	    display: block;
    }
}