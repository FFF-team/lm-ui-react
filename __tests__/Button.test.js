import React from 'react';
import Button from '../src/Button';
import { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';

/*
 * 测试点（渲染，初始化，交互）
 * 1.render结构是否正常
 * 2.检查点击事件，反馈是否正常
 */

let onClick = sinon.spy();
const normalButton = <Button
                        size='small'
                        btnType='primary'
                        isRadius={false}
                        onClick={onClick}
                        >test</Button>;
describe('Test Button', () => {

    it('render the Button wrapper', () => {

        const wrapper = shallow(normalButton);
        const wrapperClass = wrapper.find('.lm-ui-btn');
        //第一次运行时，在__snapshots__中生成快照内容
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapperClass.length === 1);
        
    })

    it('render the Button wrapper', () => {

        const button = <Button
                        size='long'
                        btnType='colorfulHollow'
                        isRadius={true}
                        onClick={onClick}
                        >test</Button>;
        const wrapper = shallow(button);
        const wrapperClass = wrapper.find('.lm-ui-btn');
        expect(wrapperClass.length === 1);
        
    })

    it('render the Button wrapper', () => {

        const button = <Button
                        size='long'
                        btnType='grayHollow'
                        isRadius={true}
                        onClick={onClick}
                        >test</Button>;
        const wrapper = shallow(button);
        const wrapperClass = wrapper.find('.lm-ui-btn');
        expect(wrapperClass.length === 1);
        
    })

    it('button click event', () => {

        const wrapper = mount(normalButton);
        wrapper.find('.lm-ui-btn').at(0).simulate('click');
        expect(onClick.called).toBe(true);
        //之后每一次进行快照对比时，都会和保存的内容作对比
        expect(toJson(wrapper)).toMatchSnapshot();

    });
})