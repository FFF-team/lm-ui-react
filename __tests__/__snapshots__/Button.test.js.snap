// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button button click event 1`] = `
<Button
  btnType="primary"
  isDisabled={false}
  isRadius={false}
  onClick={[Function]}
  size="small"
>
  <a
    className="lm-ui-btn lm-ui-btn-primary lm-ui-btn-small "
    href="javascript:;"
    onClick={[Function]}
  >
    <img
      className="hide"
    />
    test
  </a>
</Button>
`;

exports[`Test Button render the Button wrapper 1`] = `
<a
  className="lm-ui-btn lm-ui-btn-primary lm-ui-btn-small "
  href="javascript:;"
  onClick={[Function]}
>
  <img
    className="hide"
  />
  test
</a>
`;
